source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
I tried this as well and profiled the CLC. Unfortunately just calling the constructor for the TelemetryClient (even from a derived class like this one) causes the TelemetryConfiguration class to get initialized and causes the time to be wasted. It seems the only way to avoid this time penalty is to put the checks in the classes that I already have changed.